### ISSUE TITLE
Send header and body in 1 packet?

### ISSUE THREAD

**Issue by jp99** [falconry/falcon#1110]
Hi,

it seems that when sending a reply to a post request, this reply is split into two packets ( a header part and a body part). Is there a way to force Falcon to put both parts in the same tcp packet?

I'm trying to implement a simple TR069 provisioning server, and the tested clients (Yealink and Grandstream) seem (as far as i understand from debugging) not to understand the reply when split into two packets.

Jaap

**Comment by kgriffs** [comment]
Hi @jp99, unfortunately the WSGI specification prevents Falcon from having control over how the response is sent across the wire. Have you tried experimenting with different WSGI servers to see if there is one that can do what you need?

**Comment by vytas7** [comment]
I am not a network engineer, but as I understand, there is theoretically no guarantee that your packet will remain intact during transmission anyway. Your packet is only guaranteed to stay intact if it is shorter than the minimum MTU which is only [68 bytes](https://tools.ietf.org/html/rfc791) for IPv4 (now, I know that FAPP MTUs are almost always larger than 500 bytes unless you are on some slow satellite connection in Antarctica :laughing: ).

Apart from the disclaimer above, I would be surprised if some frontend proxy server such as Nginx or HAProxy would break a buffered response into packets. How does your setup look like?

**Comment by jp99** [comment]
At this stage, i'm not using a frontend at all. 
I changed the backend to Tornado, which does return the answer as 1 packet and things seem to work now. I'm a little surprised by this behavior of the client devices and searched with Wireshark for other differences in the conversation, apart from the number of packets the answer is made up of. I did not find any other differences. 

@kgriffs and @vytas7 thanks for taking time to answer my question!
